### PR TITLE
Add support for serialize and deserialize on desktop/coreCLR

### PR DESCRIPTION
**Pull request by kpanwar**
Add support for serialize and deserialize on desktop/coreCLR per https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/179

### CODE REVIEW

File: src/MSAL.Common/TokenCacheExtensions.cs [hunk 1]
```diff
@@ -44,27 +46,52 @@ public static void SetBeforeWrite(this TokenCache tokencache, TokenCache.TokenCa
         /// <summary>
         /// 
         /// </summary>
-        /// <param name="tokencache"></param>
+        /// <param name="tokenCache"></param>
```
Comment (bgavrilMS) [inline src/MSAL.Common/TokenCacheExtensions.cs]: If you feel the summary / params are self-explanatory, it would be better to just delete them.

Comment (kpanwar) [inline src/MSAL.Common/TokenCacheExtensions.cs]: i don't know why .NET adds the param for extension methods. :( I will clean it out in the next PR where I will be removing extension methods altogether.

File: src/MSAL.Common/TokenCacheExtensions.cs [hunk 2]
```diff
@@ -44,27 +46,52 @@ public static void SetBeforeWrite(this TokenCache tokencache, TokenCache.TokenCa
...
         /// <param name="state"></param>
-        public static void Deserialize(this TokenCache tokencache, byte[] state)
+        public static void Deserialize(this TokenCache tokenCache, byte[] state)
         {
-            lock (tokencache.LockObject)
+            lock (tokenCache.LockObject)
             {
+                Dictionary<string, ICollection<string>> cacheDict = JsonHelper
+                    .DeserializeFromJson<Dictionary<string, ICollection<string>>>(state);
```
Comment (bgavrilMS) [inline src/MSAL.Common/TokenCacheExtensions.cs]: Is ICollection needed? Would IEnumerable not work ?

File: src/MSAL.PCL/Internal/Cache/TokenCacheAccessor.cs [hunk 3]
```diff
@@ -70,10 +70,15 @@ public void DeleteRefreshToken(RefreshTokenCacheItem refreshToken‪Item)
         {
             TokenCachePlugin.DeleteRefreshToken(refreshToken‪Item.GetTokenCacheKey().ToString());
         }
+        
+        public ICollection<string> GetAllAccessTokensAsString()
```
Comment (bgavrilMS) [inline src/MSAL.PCL/Internal/Cache/TokenCacheAccessor.cs]: Is there a reason for returning an ICollection here and an IList below? Seems a bit inconsistent.

**Comment by bgavrilMS** [comment]
LGTM

File: src/MSAL.PCL/Internal/Cache/TokenCacheAccessor.cs (hunk 3 above)
Comment (kpanwar) [inline src/MSAL.PCL/Internal/Cache/TokenCacheAccessor.cs]: fixed

File: src/MSAL.Common/TokenCacheExtensions.cs (hunk 2 above)
Comment (kpanwar) [inline src/MSAL.Common/TokenCacheExtensions.cs]: fixed

**Pull request merged**